compiler/AST/AstCount.cpp

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright 2004-2017 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "AstCount.h"
+
+
+AstCount::AstCount() {
+#define init_members(type) num##type = 0
+foreach_ast(init_members);
+#undef init_members
+
```

Comment:
Would be nice to put a comment here like in the other places you had to handle things not covered by the macro.